### PR TITLE
Table

### PR DESCRIPTION
**Pull request by xt-t**
Successfully implemented page restaurant vote

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java [hunk 1]
```diff
@@ -1,22 +1,114 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
 
 
 @Service
 public class DateSettingsService {
 
     private final DateSettingsRepository dateSettingsRepo;
+    private static final Log LOG = LogFactory.getLog(BackendApplication.class);   //LOG herausnehmen
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:21]:
```suggestion
    private static final Log LOG = LogFactory.getLog(DateSettingsService.class);   //LOG herausnehmen
```

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java [hunk 2]
```diff
@@ -1,22 +1,114 @@
...
 
+    public Optional<TastyDateItem> addVoteTimeItemToTastyDate(UserTimeVote timeVote, String tastyDateId) {
+
+        //Optional - existiert das SettingsItem? Fehlermeldung
+
+        //checking and updating the List of the UserVotes for the DateTimes
+        if (dateSettingsRepo.findById(tastyDateId).isEmpty()) {
+            return Optional.empty();
+        }
+        TastyDateItem test = dateSettingsRepo.findById(tastyDateId).get();
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:45]:
```suggestion
        Optional<TastyDateItem> optionalTastyDateItem = dateSettingsRepo.findById(tastyDateId);
        TastyDateItem test = optionalTastyDateItem.orElseThrow(() -> new NoSuchElementException("TastyDateItem with id: " + tastyDateId + " does not exists!"));
        
```

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java [hunk 3]
```diff
@@ -1,22 +1,114 @@
...
+            for (int i=0; i < amount.length; i++) {
+                if (checksItem[i]) {
+                    amount[i]++;
+                }
+                }
+            }
+        test.setVotingResultsForOneDate(amount);
+        dateSettingsRepo.save(test);
+
+        return dateSettingsRepo.findById(tastyDateId);
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:69]:
```suggestion
        return dateSettingsRepo.save(test);
```

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java [hunk 4]
```diff
@@ -1,22 +1,114 @@
...
+        LOG.info(settingsItem);
+        UUID uuid = UUID.randomUUID();
+        settingsItem.setTastyDateId(uuid.toString());
+        settingsItem.setTimeVotes(new ArrayList<>());
+        settingsItem.setRestaurantVotes(new ArrayList<>());
+        LOG.info(settingsItem);
         return dateSettingsRepo.save(settingsItem);
     }
 
+    public Optional<TastyDateItem> addVoteTimeItemToTastyDate(UserTimeVote timeVote, String tastyDateId) {
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:37]:
```suggestion
    public TastyDateItem addVoteTimeItemToTastyDate(UserTimeVote timeVote, String tastyDateId) {
```

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java [hunk 5]
```diff
@@ -1,22 +1,114 @@
...
+                    negativeAmount[i]++;
+                }
+            }
+        }
+        test.setPositiveVotingResultsForOneRestaurant(positiveAmount);
+        test.setNegativeVotingResultsForOneRestaurant(negativeAmount);
+        dateSettingsRepo.save(test);
+
+        return dateSettingsRepo.findById(tastyDateId);
+    }
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:106]: Se above

File: frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddRestaurantManually.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    saveRestaurantData: Function
+    resetDataInput: Function
+}
+
+export default function AppointThreeAddRestaurantManually({
+                                                         appointThree,
+                                                         saveRestaurantData,
+                                                         resetDataInput
+                                                     }: AppointThreeSearchRestaurantProps) {
+    const [open, setOpen] = React.useState(false);
```
Comment (Flooooooooooorian) [inline frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddRestaurantManually.tsx:25]:
```suggestion
    const [open, setOpen] = useState(false);
```

File: backend/src/main/java/de/neuefische/backend/model/settingsSubModel/result/UserTimeVote.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.Id;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+
+public class UserTimeVote {
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/model/settingsSubModel/result/UserTimeVote.java:12]: What is this model for? Do you use it anywhere? Names are not realy clear

**Review by Flooooooooooorian — APPROVED** [review]

**Comment by Flooooooooooorian** [comment]
Add the .DS_STORE to gitignore

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java (hunk 1 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:21]: Alright. I have learned something new. :)

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java (hunk 2 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:45]: I still have to take over the optionals for a few more methods.

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java (hunk 3 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:69]: As soon as I have implemented the optionals, I will correct this. Your suggestion isn't possible due to the returned object type.

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java (hunk 4 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:37]: Nevermind the previous comment. I have taken over this and your last suggestion.

Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:37]: I still intend to revise the service class methods though.

File: backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java (hunk 5 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/DateSettingsService.java:106]: At this point I have implemented your suggestions now.

File: frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddRestaurantManually.tsx (hunk 6 above)
Comment (xt-t) [reply frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddRestaurantManually.tsx:25]: Done!

File: backend/src/main/java/de/neuefische/backend/model/settingsSubModel/result/UserTimeVote.java (hunk 7 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/model/settingsSubModel/result/UserTimeVote.java:12]: If I got time in the upcoming days, I will revise the different names. Thanks for your helpful remark!

**Pull request merged**